### DISCUSSION TITLE
Headless not working

### DISCUSSION THREAD

**Discussion by M-Zubair10** [ultrafunkamsterdam/undetected-chromedriver#1309] [Q&A]
if self.patcher.version_main < 108:
TypeError: '<' not supported between instances of 'NoneType' and 'int'

**Comment by M-Zubair10** [comment]
self.browser = uc.Chrome(use_subprocess=True, options=options, user_data_dir=user_data_dir,
                                 driver_executable_path=ChromeDriverManager().install())

**Comment by M-Zubair10** [comment]
this works
```
driver_executable_path = ChromeDriverManager().install()
version_main = int(re.findall('[0-9]+\.', driver_executable_path)[0][:-1])
driver = uc.Chrome(use_subprocess=True, options=options,
                   driver_executable_path=driver_executable_path, version_main=version_main)
```

**Comment by ultrafunkamsterdam** [comment]
Good find!   You know you don't need all this, as it downloads Chrome driver automatically ? 😅

```python
uc.Chrome(version_main=105) #v105
uc.Chrome()  #picks latest version 
```

**Comment by M-Zubair10** [comment]
No it download every time it initialized which is not a good idea for
bandwidth as well as storage
